### ISSUE TITLE
Jupyter notebooks and Origen

### ISSUE THREAD

**Issue by info-rchitect** [Origen-SDK/origen#336]
Hi,

We have installed Jupyter notebooks capable of coding in Ruby and Python.  I am wondering if anyone else has worked with Jupyter and tried to get Origen working within the notebook?

![image](https://user-images.githubusercontent.com/26782653/56581442-9757e680-659a-11e9-9330-12cb352b1d71.png)

The general idea behind a notebook is that it is a web based interactive Ruby session that could be used for training/demos.

regards

**Comment by ginty** [comment]
Never heard of it, but looks nice, not least for future enhancement of the Origen online docs.

It might be hard to get this to work within the context of a full application, but you could build up a model and use it within the notebook in the global scope.

I just made this into a quick standalone script to verify it:

~~~ruby
#!/usr/bin/env ruby
require 'origen'

class MyDut
  include Origen::TopLevel

  def initialize
    add_reg :my_reg, 0x1000
  end
end

MyDut.new

puts dut.my_reg.address.to_hex
~~~

Then running it:

~~~text
$ ./my_script.rb
0x1000
~~~

From looking at the online Juypter Ruby example that should work no problem if you have the Origen gem installed.

**Comment by info-rchitect** [comment]
@ginty Agreed on the Origen docs.  I am going to try and get this working with a full application via some user selections (application, product, silicon revision, etc.).  Stay tuned...
